File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        var left = head;
        var right = head;

        for (int i = 0; i < n; i++) {
            if (right == null) {
                return head;
            }
            right = right.next;
        }

        if (right == null) {
            return left.next;
        }
        
        while (right != null && right.next != null) {
            left = left.next;
            right = right.next;
        }
        
        if (left.next != null) {
            left.next = left.next.next;
        }
        return head;
    }
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    depth = 0
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        self.calDepth(root, 0)
        return self.depth

    def calDepth(self, root, curDepth):
        if root == None:
            return

        curDepth += 1
        if curDepth > self.depth:
            self.depth = curDepth
        
        self.calDepth(root.left, curDepth)
        self.calDepth(root.right, curDepth)
        
